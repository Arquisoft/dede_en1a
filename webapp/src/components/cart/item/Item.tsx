import { IndeterminateCheckBox, AddBox, Delete } from '@mui/icons-material';
import { useContext } from 'react';
import { Button } from 'react-bootstrap';
import { CartContext } from '../../../context/CartContext';
import './item-styles.css';
import {CartItem} from "../../../shared/shareddtypes";

type Props = {
    item: CartItem
}

const Item = ({ item }: Props) => {

    const { dispatch } = useContext(CartContext);
    let imgId = item.image === undefined ? 'undefined':item.image.toString();

    return (
        <>
            <div className='shopping-cart-item'>
                    <div className="item-action">
                    <Button
                        variant='secondary'
                        onClick={() => dispatch({
                            payload: item,
                            type: 'ADD'
                        })}>
                        <AddBox/>
                    </Button>
                    <span>{item.amount}</span>
                    <Button
                        variant='secondary'
                        onClick={() => dispatch({
                            payload: item._id,
                            type: 'REMOVE'
                        })}>
                        <IndeterminateCheckBox/>
                    </Button>
                </div>
                <div className="item-detail">
                    <div className="item-detail-image">
                        <img src={"./images/".concat(imgId).concat(".jpg")} alt={item.name} />
                    </div>
                    <div className="item-detail-info">
                        <div className="item-detail-info-name">
                            {item.name}
                        </div>
                        <div className="item-detail-info-prices">
                            <span>{item.price}â‚¬</span>
                        </div>
                    </div>
                    <div className="item-action-remove">
                        <Button
                            variant='danger'
                            title='Remove product'
                            onClick={() => dispatch({
                                payload: item._id,
                                type: 'REMOVE-ALL'
                            })}>
                            <Delete/>
                        </Button>
                    </div>
                </div>

            </div>
        </>
    )
}

export default Item;