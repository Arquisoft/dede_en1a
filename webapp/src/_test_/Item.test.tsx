import React from 'react';
import ReactDOM from 'react-dom';
import Router from "react-router-dom";
import { render, screen, cleanup } from '@testing-library/react';
import Item from '../components/cart/item/Item';

const testProduct = {
    _id: "622fc1418aedec1b7f536775",
    name: "Screwdriver",
    price: 2,
    description: "Multi-purpose screwdriver.",
    image: "b1cb26b99296eb1a8c3a912987c4a8fb5f99efe66e894d484449fe5fb55a9efb",
    weight: 1,
    amount: 10,
  }

test("Items contains product name"), () => {
  const fakeItem = render(<Item item={testProduct} />);
  expect(fakeItem).toContain(testProduct.name)
}

test("Items contains product price"), () => {
  const fakeItem = render(<Item item={testProduct} />);
  expect(fakeItem).toContain(testProduct.price)
}

test("Items contains product price"), () => {
  const fakeItem = render(<Item item={testProduct} />);
  expect(fakeItem).toContain(testProduct.price)
}

test("Item matches snapshot", () => {
    const fakeItem = render(<Item item={testProduct} />);
    expect(fakeItem).toMatchSnapshot();
 })

 //more tests -> test that correct names and descriptions show up
