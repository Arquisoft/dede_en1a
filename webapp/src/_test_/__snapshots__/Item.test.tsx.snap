// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="shopping-cart-item"
      >
        <div
          class="item-action"
        >
          <button
            class="btn btn-secondary"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
          </button>
          <span>
            10
          </span>
          <button
            class="btn btn-secondary"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="IndeterminateCheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
              />
            </svg>
          </button>
        </div>
        <div
          class="item-detail"
        >
          <div
            class="item-detail-image"
          >
            <img
              alt="Screwdriver"
              src="./images/b1cb26b99296eb1a8c3a912987c4a8fb5f99efe66e894d484449fe5fb55a9efb.jpg"
            />
          </div>
          <div
            class="item-detail-info"
          >
            <div
              class="item-detail-info-name"
            >
              Screwdriver
            </div>
            <div
              class="item-detail-info-prices"
            >
              <span>
                2
                 €
              </span>
            </div>
          </div>
          <div
            class="item-action-remove"
          >
            <button
              class="btn btn-danger"
              title="Remove product"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="shopping-cart-item"
    >
      <div
        class="item-action"
      >
        <button
          class="btn btn-secondary"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
        </button>
        <span>
          10
        </span>
        <button
          class="btn btn-secondary"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="IndeterminateCheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
            />
          </svg>
        </button>
      </div>
      <div
        class="item-detail"
      >
        <div
          class="item-detail-image"
        >
          <img
            alt="Screwdriver"
            src="./images/b1cb26b99296eb1a8c3a912987c4a8fb5f99efe66e894d484449fe5fb55a9efb.jpg"
          />
        </div>
        <div
          class="item-detail-info"
        >
          <div
            class="item-detail-info-name"
          >
            Screwdriver
          </div>
          <div
            class="item-detail-info-prices"
          >
            <span>
              2
               €
            </span>
          </div>
        </div>
        <div
          class="item-action-remove"
        >
          <button
            class="btn btn-danger"
            title="Remove product"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
